rt/bat/RUNOPTABLES.BAT

```diff
@@ -14,88 +14,283 @@ rem ----- Lvl(%z_NestLevel%) Start %0 %1 %2 %3 %4 %5 %6 %7 %8 %9
+                         )
+if %1. NEQ . goto return
+
+rem  Machine ARCH-1 - equivalent of Optable(ESA)
+:ARCH-1
+set Z390PRN=rt\mlc\OPTB_ARCH-1.PRN
+set HLASMPRN=rt\mlc\OPTB#ESA.TF1
+call bat\asmlg.bat %z_TraceMode% rt\mlc\OPTB# @rt\mlc\OPTB_ARCH-1.OPT sysprn(%Z390PRN%)
+set z_ReturnCode=%ERRORLEVEL%
+if %z_ReturnCode% EQU 0 (echo %0 - OPTB# "machine(ARCH-1)" is okay
+                         set /A count_ok=%count_ok%+1
+                         ) else (
+                         echo %0 ERROR: OPTB# returned %z_ReturnCode% for "machine(ARCH-1)"
+                         set /A count_err=%count_err%+1
+                         if (%z_ReturnCode% GTR %z_MaxRetCode% (set z_MaxRetCode=%z_ReturnCode%)
+                         )
+if %1. NEQ . goto return
+
+rem  Machine ARCH-2 - equivalent of Optable(ESA)
+:ARCH-2
+set Z390PRN=rt\mlc\OPTB_ARCH-2.PRN
+set HLASMPRN=rt\mlc\OPTB#ESA.TF1
+call bat\asmlg.bat %z_TraceMode% rt\mlc\OPTB# @rt\mlc\OPTB_ARCH-2.OPT sysprn(%Z390PRN%)
+set z_ReturnCode=%ERRORLEVEL%
+if %z_ReturnCode% EQU 0 (echo %0 - OPTB# "machine(ARCH-2)" is okay
+                         set /A count_ok=%count_ok%+1
+                         ) else (
+                         echo %0 ERROR: OPTB# returned %z_ReturnCode% for "machine(ARCH-2)"
+                         set /A count_err=%count_err%+1
                          if (%z_ReturnCode% GTR %z_MaxRetCode% (set z_MaxRetCode=%z_ReturnCode%)
                          )
+if %1. NEQ . goto return
+
+rem  Machine ARCH-3 - equivalent of Optable(ESA)
+:ARCH-3
+set Z390PRN=rt\mlc\OPTB_ARCH-3.PRN
+set HLASMPRN=rt\mlc\OPTB#ESA.TF1
+call bat\asmlg.bat %z_TraceMode% rt\mlc\OPTB# @rt\mlc\OPTB_ARCH-3.OPT sysprn(%Z390PRN%)
+set z_ReturnCode=%ERRORLEVEL%
+if %z_ReturnCode% EQU 0 (echo %0 - OPTB# "machine(ARCH-3)" is okay
+                         set /A count_ok=%count_ok%+1
+                         ) else (
+                         echo %0 ERROR: OPTB# returned %z_ReturnCode% for "machine(ARCH-3)"
+                         set /A count_err=%count_err%+1
+                         if (%z_ReturnCode% GTR %z_MaxRetCode% (set z_MaxRetCode=%z_ReturnCode%)
+                         )
+if %1. NEQ . goto return
+
+rem  Machine ARCH-4 - equivalent of Optable(ESA)
+:ARCH-4
+set Z390PRN=rt\mlc\OPTB_ARCH-4.PRN
+set HLASMPRN=rt\mlc\OPTB#ESA.TF1
+call bat\asmlg.bat %z_TraceMode% rt\mlc\OPTB# @rt\mlc\OPTB_ARCH-4.OPT sysprn(%Z390PRN%)
+set z_ReturnCode=%ERRORLEVEL%
+if %z_ReturnCode% EQU 0 (echo %0 - OPTB# "machine(ARCH-4)" is okay
+                         set /A count_ok=%count_ok%+1
+                         ) else (
+                         echo %0 ERROR: OPTB# returned %z_ReturnCode% for "machine(ARCH-4)"
+                         set /A count_err=%count_err%+1
+                         if (%z_ReturnCode% GTR %z_MaxRetCode% (set z_MaxRetCode=%z_ReturnCode%)
+                         )
+if %1. NEQ . goto return
+
+:report
+echo %count_ok% tests were okay and %count_err% failed
```

Comment:
:Lot of work here!

What follows is just a suggestion. Not required. ONLY DO THIS IF YOU WANT TO DO IT!

You could have 2 (count-ok, count-err) pairs -- one for optable, one for machine. At label :report echo the optable subtotals and echo the machine subtotals, followed by and echo of the grand totals.